test/stdlib/NSObject-hashing.swift

```diff
@@ -0,0 +1,15 @@
+// RUN: %target-typecheck-verify-swift
+// RUN: %target-swift-frontend -c -verify -verify-ignore-unknown %s -o /dev/null
+
+// REQUIRES: objc_interop
+
+import ObjectiveC
+
+class Foo: NSObject {
+  override var hashValue: Int { // expected-error {{overriding non-open property outside of its defining module}} expected-error {{overriding non-@objc declarations from extensions is not supported}}
+    return 0
+  }
+
+  override func hash(into hasher: inout Hasher) { // expected-error {{overriding non-open instance method outside of its defining module}} expected-error {{overriding declarations in extensions is not supported}}
+  }
+}
```

Comment:
Maybe something here should show the correct way to override `hash`?

Reply:
Agreed! The warning we have in 4.2 is extremely helpful. I have a pending task to improve error diagnostics -- it's not ABI impacting though.

```
warning: override of 'NSObject.hashValue' is deprecated; override 'NSObject.hash' to get consistent hashing behavior
```